skills/repo.go

```diff
@@ -19,10 +19,45 @@
 
 package skills
 
+import (
+	"fmt"
+	"sync"
+)
+
 // Repository stores all known snappy skills and slots and types.
-type Repository struct{}
+type Repository struct {
+	// Protects the internals from concurrent access.
+	m     sync.Mutex
+	types map[string]Type
+}
 
 // NewRepository creates an empty skill repository.
 func NewRepository() *Repository {
-	return &Repository{}
+	return &Repository{
+		types: make(map[string]Type),
+	}
+}
+
+// Type returns a type with a given name.
+func (r *Repository) Type(typeName string) Type {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	return r.types[typeName]
+}
+
+// AddType adds the provided skill type to the repository.
+func (r *Repository) AddType(t Type) error {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	typeName := t.Name()
+	if err := ValidateName(typeName); err != nil {
+		return err
+	}
+	if _, ok := r.types[typeName]; ok {
+		return fmt.Errorf("cannot add duplicated type name: %q", typeName)
```

Comment:
Indeed, or even better: `"cannot add skill type: %q type name is in use"`

Sorry for the bikeshed, @zyga. It's probably worth improving the message, at least to point out that this is a _skill_ type, which I missed in my first suggestion.